### PR TITLE
fix: replace None return type of gRPC request handlers

### LINKED ISSUE #2063: run_async returns future - not None
crud functions in `jina/clients/__init__.py` have the None annotation. However, they return a future.

please make sure the return type is correct

### PR DESCRIPTION
**Pull request by Kelton8Z**
close #2063

### CODE REVIEW

File: jina/clients/__init__.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def train(
         on_error: CallbackFnType = None,
         on_always: CallbackFnType = None,
         **kwargs
-    ) -> None:
+    ) -> Awaitable:
```
Comment (florian-hoenicke) [inline jina/clients/__init__.py:44]: you can define the type of the `Awaitable`. like `Awaitable[...]`

File: jina/clients/__init__.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def search(
...
```
Comment (florian-hoenicke) [inline jina/clients/__init__.py:72]: you can check if other return types are possible as well.

**Review by florian-hoenicke — CHANGES_REQUESTED** [review]

**Comment by florian-hoenicke** [comment]
@Kelton8Z There are other places as well where the type annotation is needed `clients/asyncio`, `jina/flow/mixin/crud.py ...

File: jina/clients/__init__.py [hunk 3]
```diff
@@ -98,7 +98,7 @@ def index(
...
```
Comment (florian-hoenicke) [inline jina/clients/__init__.py:101]: @Kelton8Z Is `Awaitalbe` the correct type? That is the sync client - not the async variation.

**Review by florian-hoenicke — CHANGES_REQUESTED** [review]

File: jina/clients/__init__.py (hunk 3 above)
Comment (Kelton8Z) [reply jina/clients/__init__.py:101]: is index a sync client in the sense that the flow needs to finish indexing before moving on to query or other process?

in that case i also need to change the code within the index function ?

**Review by JoanFM — CHANGES_REQUESTED** [review]
Lets separate the 2 PR. they are different stories

**Review by JoanFM — CHANGES_REQUESTED** [review]
@Kelton8Z watch out because u have changes in the hub submodule

File: jina/clients/__init__.py [hunk 4]
```diff
@@ -126,15 +126,15 @@ def update(
         on_error: CallbackFnType = None,
         on_always: CallbackFnType = None,
         **kwargs
-    ) -> Awaitable:
+    ) -> Awaitable[list]:
```
Comment (JoanFM) [inline jina/clients/__init__.py:129]: capital Letter. Do we know whay is the list of? is a list of Documents?

Comment (Kelton8Z) [reply jina/clients/__init__.py:129]: oh yeah, a list of responses

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Comment by Kelton8Z** [comment]
> @Kelton8Z watch out because u have changes in the hub submodule

"The behaviour of git commit -a seems to be to include changes to submodules, if they have new commits inside them." is probably how I introduced them accidentally.

**Comment by JoanFM** [comment]
> > @Kelton8Z watch out because u have changes in the hub submodule
> 
> "The behaviour of git commit -a seems to be to include changes to submodules, if they have new commits inside them." is probably how I introduced them accidentally.

do not use -a, just select what u want to add

File: jina/clients/asyncio.py [hunk 5]
```diff
@@ -93,7 +94,7 @@ async def search(
         on_error: CallbackFnType = None,
         on_always: CallbackFnType = None,
         **kwargs
-    ) -> None:
+    ) -> AsyncGenerator:
```
Comment (JoanFM) [inline jina/clients/asyncio.py:97]: and here?

File: jina/clients/__init__.py [hunk 6]
```diff
@@ -126,7 +126,7 @@ def update(
         on_error: CallbackFnType = None,
         on_always: CallbackFnType = None,
         **kwargs
-    ) -> None:
+    ) -> Awaitable:
```
Comment (JoanFM) [inline jina/clients/__init__.py:129]: Awaiteble of something?

**Comment by nan-wang** [comment]
duplicated with #2386

**Pull request closed without merging**